src/cmds.c

```diff
@@ -1907,13 +1907,13 @@ static int add_to_handle(struct userrec *u, int idx, char *handle, char *host, i
       return 1;
     }
   }
-  if ( !type && !glob_botmast(fr) && !chan_master(fr) && get_user_by_host(host)) {
+  if ( !type && !glob_botmast(fr) && !chan_master(fr) && lookup_user_record(NULL, NULL, host)) {
     dprintf(idx, "You cannot add %s matching another user!\n",
```

Comment:
the type condition on that dprintf is unnecessary, the if-condition already includes it, we're guaranteed to be dealing with  hostmask